engine/src/main/java/org/terasology/engine/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -526,7 +526,8 @@ public <T extends Component> T addComponent(long entityId, T component) {
         if (!oldComponent.isPresent()) {
             notifyComponentAdded(getEntity(entityId), component.getClass());
         } else {
-            logger.error("Adding a component ({}) over an existing component for entity {}", component.getClass(), entityId);
+            logger.atError().addArgument(() -> component.getClass()).addArgument(() -> entityId).
+                    log("Adding a component ({}) over an existing component for entity {}");
```

Comment:
`getClass()` should be pretty cheap so impact can be considered neglectable
```suggestion
            logger.error("Adding a component ({}) over an existing component for entity {}", component.getClass(), entityId); //NOPMD
```